airbyte-integrations/bases/base-normalization/normalization/transform_catalog/destination_name_transformer.py

```diff
@@ -0,0 +1,188 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import unicodedata as ud
+from re import match, sub
+
+from normalization.destination_type import DestinationType
+from normalization.transform_catalog.reserved_keywords import is_reserved_keyword
+from normalization.transform_catalog.utils import jinja_call
+
+DESTINATION_SIZE_LIMITS = {
+    DestinationType.BIGQUERY.value: 1024,
+    DestinationType.SNOWFLAKE.value: 255,
+    DestinationType.REDSHIFT.value: 127,
+    DestinationType.POSTGRES.value: 63,
+}
+
+# DBT also needs to generate suffix to table names, so we need to make sure it has enough characters to do so...
+TRUNCATE_DBT_RESERVED_SIZE = 12
+
+# We reserve this many characters from identifier names to be used for prefix/suffix for airbyte
+# before reaching the database name length limit
+# 2 characters for signaling truncate with '__' and 6 others for generating unique strings
+TRUNCATE_RESERVED_SIZE: int = 8
+
+
+class DestinationNameTransformer:
+    """
+    Handles naming conventions in destinations for all kind of sql identifiers:
+    - schema
+    - table
+    - column
+    """
+
+    def __init__(self, integration_type: DestinationType):
+        """
+        @param integration_type is the destination type of warehouse
+        """
+        self.integration_type: DestinationType = integration_type
+
+    # Public methods
+
+    def needs_quotes(self, input_name: str) -> bool:
+        """
+        @param input_name to test if it needs to manipulated with quotes or not
+        """
+        if is_reserved_keyword(input_name, self.integration_type):
+            return True
+        if self.integration_type.value == DestinationType.BIGQUERY.value:
+            return False
+        doesnt_start_with_alphaunderscore = match("[^A-Za-z_]", input_name[0]) is not None
```

Comment:
why not: `doesnt_start_with_alphaunderscore = match("^[^A-Za-z_]", input_name) is not None`